### PR TITLE
New sampler report forkmat provided in PR 40 on sampler project

### PR DESCRIPTION
**Pull request by mtodor**

### CODE REVIEW

**Comment by chrfritsch** [comment]
Testsite builder throws this error on config creation:

Error: Call to a member function id() on null in /Users/d430774/Sites/drupal/core/modules/shortcut/src/Entity/ShortcutSet.php on line 78 #0 /Users/d430774/Sites/drupal/core/lib/Drupal/Core/Entity/EntityStorageBase.php(526): Drupal\shortcut\Entity\ShortcutSet->postSave(Object(Drupal\shortcut\ShortcutSetStorage), false)
#1 /Users/d430774/Sites/drupal/core/lib/Drupal/Core/Entity/EntityStorageBase.php(452): Drupal\Core\Entity\EntityStorageBase->doPostSave(Object(Drupal\shortcut\Entity\ShortcutSet), false)
#2 /Users/d430774/Sites/drupal/core/lib/Drupal/Core/Config/Entity/ConfigEntityStorage.php(263): Drupal\Core\Entity\EntityStorageBase->save(Object(Drupal\shortcut\Entity\ShortcutSet))
#3 /Users/d430774/Sites/drupal/core/lib/Drupal/Core/Entity/EntityBase.php(394): Drupal\Core\Config\Entity\ConfigEntityStorage->save(Object(Drupal\shortcut\Entity\ShortcutSet))
#4 /Users/d430774/Sites/drupal/core/lib/Drupal/Core/Config/Entity/ConfigEntityBase.php(613): Drupal\Core\Entity\EntityBase->save()
#5 /Users/d430774/Sites/drupal/modules/contrib/testsite_builder/src/Plugin/testsite_builder/EntityType/EntityTypeBase.php(51): Drupal\Core\Config\Entity\ConfigEntityBase->save()
#6 /Users/d430774/Sites/drupal/modules/contrib/testsite_builder/src/ConfigCreator.php(162): Drupal\testsite_builder\Plugin\testsite_builder\EntityType\EntityTypeBase->createBundle('bundle_0', Array)
#7 /Users/d430774/Sites/drupal/modules/contrib/testsite_builder/src/Command/CreateConfigCommand.php(105): Drupal\testsite_builder\ConfigCreator->create()
#8 /Users/d430774/Sites/drupal/vendor/symfony/console/Command/Command.php(255): Drupal\testsite_builder\Command\CreateConfigCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#9 /Users/d430774/Sites/drupal/vendor/symfony/console/Application.php(978): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#10 /Users/d430774/Sites/drupal/vendor/symfony/console/Application.php(255): Symfony\Component\Console\Application->doRunCommand(Object(Drupal\testsite_builder\Command\CreateConfigCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /Users/d430774/Sites/drupal/vendor/drupal/console-core/src/Application.php(187): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 /Users/d430774/Sites/drupal/vendor/drupal/console/src/Application.php(64): Drupal\Console\Core\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /Users/d430774/Sites/drupal/vendor/symfony/console/Application.php(148): Drupal\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /Users/d430774/Sites/drupal/vendor/drupal/console/bin/drupal.php(89): Symfony\Component\Console\Application->run()
#15 /Users/d430774/Sites/drupal/vendor/drupal/console/bin/drupal(4): require('/Users/d430774/...')
#16 {main}
Error: Call to a member function id() on null in Drupal\shortcut\Entity\ShortcutSet->postSave() (line 78 of /Users/d430774/Sites/drupal/core/modules/shortcut/src/Entity/ShortcutSet.php).

Here is my config:
[report.json.zip](https://github.com/thunder/testsite_builder/files/3446139/report.json.zip)

File: src/ConfigCreator.php [hunk 1]
```diff
@@ -163,10 +163,9 @@ public function create(): ConfigCreator {
         $this->eventDispatcher->dispatch(ConfigCreatorEvents::ENTITY_BUNDLE_CREATE, new ConfigCreatorEntityBundleCreateEvent($bundle, $bundle_config));
         $configuration['bundle_type'] = $bundle_id;
 
-        // Create fields.
-        $created_fields = [];
-        foreach ($bundle_config['fields'] as $field_index => $field_instance) {
-          $field_configuration = $configuration + $field_instance;
+        foreach ($bundle_config['fields'] as $field_name => $field_instance) {
+          $field_name = str_replace('-', '_', $field_name);
```
Comment (chrfritsch) [inline src/ConfigCreator.php]: Are there cases where the field_name can contain a - ?

**Review by chrfritsch — CHANGES_REQUESTED** [review]
Code looks good. Just one question. But still getting that error.

File: src/ConfigCreator.php (hunk 1 above)
Comment (mtodor) [reply src/ConfigCreator.php]: No, it was initially in PR, but then I have asked Daniel to change that into underline, so I should remove that. 👍

**Review by chrfritsch — APPROVED** [review]

**Pull request merged**